validator/api/src/main/java/tech/pegasys/teku/validator/api/MultimodeGraffitiProvider.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Consensys Software Inc., 2025
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.api;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.bls.BLSPublicKey;
+
+/**
+ * A GraffitiProvider that supports multiple graffiti entries in various formats: - specific: Allows
+ * setting specific graffiti for individual validators by index or public key - ordered: Use
+ * graffiti messages in the specified order, cycling through them - random: Randomly select graffiti
+ * from a list - default: Fallback graffiti when no other option applies
+ */
+public class MultimodeGraffitiProvider implements GraffitiProvider {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final Random RANDOM = new Random();
+
+  private final Optional<Bytes32> defaultGraffiti;
+  private final Optional<BLSPublicKey> validatorPublicKey;
+  private final Optional<Path> graffitiFile;
+  private final AtomicInteger orderedIndex = new AtomicInteger(0);
+
+  private volatile Optional<GraffitiConfiguration> graffitiConfig = Optional.empty();
+  private volatile long lastFileCheckTime = 0;
+  private volatile long lastFileModifiedTime = 0;
+
+  public MultimodeGraffitiProvider(
+      final Optional<Bytes32> defaultGraffiti,
+      final Optional<BLSPublicKey> validatorPublicKey,
+      final Optional<Path> graffitiFile) {
+    this.defaultGraffiti = defaultGraffiti;
+    this.validatorPublicKey = validatorPublicKey;
+    this.graffitiFile = graffitiFile;
+  }
+
+  @Override
+  public Optional<Bytes32> get() {
```

Comment:
We need to make sure this method can never interfere with block production. While I was testing it, I noticed that any errors were bubbling up and not being handled here. The result was a failure in block production!

We need to ensure that any errors that happen inside `get()` are caught and the method returns Optional.empty().

One example on how I managed to trigger this was to set `defaultGraffiti:` with a value larger then 32 bytes.